packages/metadata/tests/productV1Item.test.ts

```diff
@@ -0,0 +1,381 @@
+        )}`
+      );
+    });
+
+    test("should fail if some variants have the same option", async () => {
+      const productItem = productV1ItemValidMinimal as unknown as ProductV1Item;
+      const variants = cloneVariants(variantsOK);
+      // Set the last variant equal to the first variant
+      variants[variants.length - 1].productVariant = [
+        ...variants[0].productVariant
+      ];
+      expect(() => createVariantProductItem(productItem, variants)).toThrow(
+        `Redundant variant ${serializeVariant(variants[0].productVariant)}`
+      );
+    });
+
+    test("should fail if some variants have the same option - disordered variants table", async () => {
+      const productItem = productV1ItemValidMinimal as unknown as ProductV1Item;
+      const variants = cloneVariants(variantsOK);
+      // Set the last variant equal to the first variant
+      variants[variants.length - 1].productVariant = [
+        ...variants[0].productVariant
+      ];
+      variants[0].productVariant = disorderVariantTable(
+        variants[0].productVariant
+      );
+      expect(() => createVariantProductItem(productItem, variants)).toThrow(
+        `Redundant variant ${serializeVariant(variants[0].productVariant)}`
+      );
+    });
+
+    test("should fail if some variants have the same option - disordered variants structure", async () => {
+      const productItem = productV1ItemValidMinimal as unknown as ProductV1Item;
+      const variants = cloneVariants(variantsOK);
+      // Set the last variant equal to the first variant
+      variants[variants.length - 1].productVariant = [
+        ...variants[0].productVariant
+      ];
+      variants[0].productVariant = disorderVariantStruct(
+        variants[0].productVariant
+      );
+      expect(() => createVariantProductItem(productItem, variants)).toThrow(
+        `Redundant variant ${serializeVariant(variants[0].productVariant)}`
+      );
+    });
+
+    test("should fail if some variants have the same option (bis)", async () => {
+      // The previous test is setting the last variants equals to the first one, to create a redundancy.
+      // This other test just does the same thing but between the second and the penultimate
+      //  (last but one), just to verify the algorithm is not only working for the 1st one !
+      const productItem = productV1ItemValidMinimal as unknown as ProductV1Item;
+      const variants = cloneVariants(variantsOK);
+      // Set the penultimate variant equal to the second variant (only works when variants.length >= 4)
+      expect(variants.length).toBeGreaterThanOrEqual(4);
+      variants[variants.length - 2].productVariant = [
+        ...variants[1].productVariant
+      ];
+      expect(() => createVariantProductItem(productItem, variants)).toThrow(
+        `Redundant variant ${serializeVariant(variants[1].productVariant)}`
+      );
+    });
+
+    test("should NOT fail if the variants table is disordered ", async () => {
```

Comment:
```suggestion
    test("should NOT fail if the variants table is unordered ", async () => {
```

(same in other places)

Reply:
Done (all occurences)